macroslib/src/cpp/mod.rs

```diff
@@ -533,6 +537,17 @@ fn merge_rule(ctx: &mut CppContext, mut rule: TypeMapConvRuleInfo) -> Result<()>
     Ok(())
 }
 
+fn cache_f_code(cached_code: &mut FxHashSet<(SmolStr, String)>, f_module_name: &SmolStr, f_code_str: &str) -> bool {
+    for (module_name, code) in cached_code.iter() {
+        if (module_name == f_module_name) && (code.as_str() == f_code_str) {
+            return true;
+        }
+    }
+
+    cached_code.insert((f_module_name.clone(), String::from(f_code_str)));
```

Comment:
You can write f_code_str.into(), this is shorter, and survive if type of f_code_str would be changed in the future.

Reply:
`f_code_str.into()` converts to `UniqueName`, then, changed to `f_code_str.to_string()`